### PR TITLE
FIX: add support for sidebar tags list

### PR DESCRIPTION
**Pull request by fzngagan**

### CODE REVIEW

**Review by CvX — COMMENTED** [review]
Not a fan of modifying a class (/its prototype) directly, but it's a niche functionality so maybe it doesn't warrant a proper plugin api… And the way we run theme component tests prevents cross-contamination so maybe it's okay. For now. 😅

File: javascripts/discourse/api-initializers/remove_dashes_from_tag_names.js
[the remark this answers is not in the document]
Comment (fzngagan) [reply javascripts/discourse/api-initializers/remove_dashes_from_tag_names.js:1]: @CvX I've renamed the file to use `-`. The filename is a result of `discourse_theme new ` command. Maybe we should also fix that there?

 Also, I'm thinking to make the `domFromString` change in a separate PR. Sounds good?

**Comment by fzngagan** [comment]
> Not a fan of modifying a class (/its prototype) directly, but it's a niche functionality so maybe it doesn't warrant a proper plugin api… And the way we run theme component tests prevents cross-contamination so maybe it's okay. For now. 😅

And, since we're in discussion about making a new api to replace the `modifyClass` thingy, I didn't want to introduce a new api yet.

File: javascripts/discourse/api-initializers/remove_dashes_from_tag_names.js
[the remark this answers is not in the document]
Comment (CvX) [reply javascripts/discourse/api-initializers/remove_dashes_from_tag_names.js:1]: 👍 for the separate PR!

re: `discourse_theme new` - I think it's either already fixed or the name entered then used underscores, and I'd rather not introduce automatic correction there (I'm sure somebody would then come up with a case where they **need** an underscore in their name 😛)

File: javascripts/discourse/api-initializers/remove-dashes-from-tag-names.js [hunk 1]
```diff
@@ -18,4 +19,11 @@ export default apiInitializer("0.11.1", (api) => {
   };
 
   api.replaceTagRenderer(customRenderer);
+
+  Object.defineProperty(BaseTagSectionLink.prototype, "text", {
+    get() {
+      let name = this.name;
+      return name.replace(/-/g, " ");
```
Comment (CvX) [inline javascripts/discourse/api-initializers/remove-dashes-from-tag-names.js:26]: golf still 😛 

```suggestion
      return this.name.replace(/-/g, " ");
```

**Review by CvX — APPROVED** [review]

**Pull request merged**